### DISCUSSION TITLE
Using command URI's in a markdown document via TextDocumentProvider

### DISCUSSION THREAD

**Discussion by caleb-at-pieces** [microsoft/vscode-discussions#1078] [Extension Development QnA] [answered]
Hello,

My extension uses the `vscode.TextDocumentProvider` api in order to provide previews of a user's saved material. I would like to add some buttons that link to command URI's in order to perform some actions in this view but it seems that this may be unsupported. Is there a way for me to add functional links to command uri's within the content that I provide within a TextDocumentProvider?

i.e:

```markdown
[Command](command:vscode.command.command)
```

**Comment by gjsjohnmurray** [comment, marked as answer]
Did you look at implementing a [`DocumentLinkProvider`](https://code.visualstudio.com/api/references/vscode-api#DocumentLinkProvider&lt;T&gt;)?

**Reply by caleb-at-pieces** [reply]
Looks like this should work for me! I will give this a shot. Thank you

**Comment by caleb-at-pieces** [comment]
> Did you look at implementing a [`DocumentLinkProvider`](https://code.visualstudio.com/api/references/vscode-api#DocumentLinkProvider%3CT%3E)?

I have looked into this however it looks like I am not able to use this as I am using the 'markdown.showPreview' command in order to show my provided text documents.

```typescript
const uri = mycustomuri;
vscode.commands.executeCommand('markdown.showPreview', uri);
```
